.github/workflows/ci.yml

```
@@ -31,7 +31,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        example-code: ["include_example_code=yes", "include_example_code=no"]
+        ejections:
```

Changing to this matrix structure makes it more difficult to add additional options in the future. Also, it's not obvious what the values of the non-set options are. Finally, this is running 4 times, but two of the iterations actually yield the same values.

I think this would be better:

```suggestion
        include:
          - example-code: "include_example_code=no"
            asgi: "use_asgi=no"
          - example-code: "include_example_code=yes"
            asgi: "use_asgi=no"
          - example-code: "include_example_code=yes"
            asgi: "use_asgi=yes"
```

I'm fine to intentionally omit the case of (no,yes), since the interaction between these two options is limited.

We'll just need to add `--data ${{ matrix.asgi }}` to the command below too.